akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala

```diff
@@ -33,11 +33,24 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
     new Source(delegate.mergeSubstreams)
 
   /**
+   * Flatten the sub-flows back into the super-flow by performing a merge
+   * with the given breadth limit. This means that only up to `breadth`
+   * substreams will be executed at any given time. Substreams that are not
+   * yet executed are also not materialized, meaning that back-pressure will
+   * be exerted at the operator that creates the substreams when the breadth
+   * limit is reached.
+   */
+  def mergeSubstreamsWithLimit(breadth: Int): Source[Out, Mat] =
```

Comment:
since mapAsync isn't named mapAsyncWithParallelism I think it's fine to not
name it WithParallelism.

If you are worried about the overload then I wouldn't worry.

-- 
Cheers,
√
On 12 Dec 2015 17:38, "kerr" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala
> <https://github.com/akka/akka/pull/19148#discussion_r47433632>:
>
> > @@ -33,11 +33,24 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
> >      new Source(delegate.mergeSubstreams)
> >
> >    /**
> > +   * Flatten the sub-flows back into the super-flow by performing a merge
> > +   * with the given breadth limit. This means that only up to `breadth`
> > +   * substreams will be executed at any given time. Substreams that are not
> > +   * yet executed are also not materialized, meaning that back-pressure will
> > +   * be exerted at the operator that creates the substreams when the breadth
> > +   * limit is reached.
> > +   */
> > +  def mergeSubstreamsWithLimit(breadth: Int): Source[Out, Mat] =
>
> I thin we could drop the WithParallelism.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19148/files#r47433632>.
>